src/server/scripts/EasternKingdoms/zone_elwynn_forest.cpp

```diff
@@ -275,7 +275,263 @@ struct npc_cameron : public ScriptedAI
     GuidVector _childrenGUIDs;
 };
 
+/*######
+## npc_supervisor_raelen
+######*/
+
+enum SupervisorRaelen
+{
+    EVENT_FIND_PEASANTS  = 1,
+    EVENT_NEXT_PEASANT   = 2,
+    NPC_EASTVALE_PEASANT = 11328
+};
+
+struct npc_supervisor_raelen : public ScriptedAI
+{
+    npc_supervisor_raelen(Creature* creature) : ScriptedAI(creature)
+    {
+        _PeasantId = 0;
+    }
+
+    void Reset() override
+    {
+        _PeasantId = 0;
+        peasantGUIDs.clear();
+        _events.ScheduleEvent(EVENT_FIND_PEASANTS, 4s);
+    }
+
+    void SetData(uint32 /*type*/, uint32 data) override
+    {
+        if (data == 1)
+        {
+            ++_PeasantId;
+            if (_PeasantId == 5) _PeasantId = 0;
+            _events.ScheduleEvent(EVENT_NEXT_PEASANT, 2s, 6s);
+        }
+    }
+
+    void CallPeasent()
+    {
+        if (Creature* peasant = ObjectAccessor::GetCreature(*me, peasantGUIDs[_PeasantId]))
+            peasant->AI()->SetData(1, 1);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        if (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_FIND_PEASANTS:
+                {
+                    GuidVector tempGUIDs;
+                    std::list<Creature*> peasants;
+                    GetCreatureListWithEntryInGrid(peasants, me, NPC_EASTVALE_PEASANT, 100.f);
```

Comment:
you need to check whether container peasants is not empty, otherwise the worldserver will crash when you iterate through it and it's empty.

Reply:
AN if (peasants) for lines 332 to 341?

Reply:
as Shauren pointed GetCreatureListWithEntryInGrid won't guarantee that the list you're providing below will be the same order you want it, for that purpose you'd need to sort it first, but, to do so, you'd need StringId, for example, which isn't ported from Master currently.

also, if GetCreatureListWithEntryInGrid fails to obtain the exact number you're referencing, in this case a container of 5 elements, the code will crash too sadly, as I mentioned earlier

Reply:
I see an issue with code. will work on it tonight.